templates/service.mustache

```diff
@@ -5,7 +5,7 @@
 /* eslint-disable  @typescript-eslint/no-empty-interface */
 /* eslint-disable  @typescript-eslint/no-unused-vars */
 import * as ApiModelTypes from "{{{TYPES_DIRECTORY}}}";
-import axios, { AxiosResponse, CancelToken{{^IS_AXIOS_VERSION_ZERO}}, RawAxiosRequestHeaders{{/IS_AXIOS_VERSION_ZERO}} } from "axios";
+import axios, { AxiosResponse, RawAxiosRequestHeaders{{/IS_AXIOS_VERSION_ZERO}} } from "axios";
```

Comment:
So I'm observing an error when trying to generate the files. 
![image](https://github.com/AlanCea68/tag-api-codegen/assets/83241918/ea9f5edd-4bce-4cd1-91ed-9422327cc524)

---
So `{{/IS_AXIOS_VERSION_ZERO}}` is a mustache tag that defines the end of a "section" indicated by the slash `/`. Think of this like the closing tag of a JSX/TSX element "`</component>`".

A "section" in mustache is a block of template between an opening and a closing tag that is conditionally rendered with the conditional logic being based off the opening tag for which there are two types. 



- First, a normal section tag which starts with a pound symbol `{{#KEY}}`, which means if the key does not exist, or exists and has a value of null, undefined, false, 0, or NaN, or is an empty string or an empty list, the block will not be rendered.
- Second, an inverted section tag which starts with the carrot symbol `{{^KEY}}`, which means the block of an inverted section is rendered only if the value of that section's tag is null, undefined, false, falsy or an empty list. 

In the case before this modification the opening tag was `{{^IS_AXIOS_VERSION_ZERO}}` which has been removed, this is a violation of mustache rules and hence the error. You can learn more about how the mustache template works visiting this page: https://github.com/janl/mustache.js and scrolling down. 


What we need to figure out is if we need to keep the inverted opening tag or put in the normal opening tag. 

It looks to me like based off the changes then it would still be ok for us to keep the inverted tag as the abort controller is supported prior to v1 which is the value for which `IS_AXIOS_VERSION_ZERO` is based on. 

Tagging @kellirains for inputs here so whatever we do here aligns with her recent changes with supporting differing axios versions. Can you help review to make sure the changes here doesn't conflict with your recent logic?

Reply:
Sure! My changes with the IS_AXIOS_VERSION_ZERO variable added a conditional type RawAxiosRequestHeaders when the user provides an Axios version and it is NOT zero, as it's unsupported in 0.x versions but versions > 1.0 give a type error when using `Record<string, unknown>` (the previous type for headers). You can see the conditional headers types being generated on line 57.

So we do need the entire block `{{^IS_AXIOS_VERSION_ZERO}} RawAxiosRequestHeaders{{/IS_AXIOS_VERSION_ZERO}}` and it needs to be inverted, which adds `, RawAxiosRequestHeaders` into the import statement if the version is NOT zero.

Reply:
This makes SO much sense now! Thanks team.